src/Autosuggest.js

```diff
@@ -90,6 +92,15 @@ export default class Autosuggest extends Component {
         this.handleValueChange(nextProps.value);
       }
     }
+    if (nextProps.debounce !== this.props.debounce || nextProps.suggestions !== this.props.suggestions) {
+      this.updateSuggestionsFn(nextProps);
+    }
```

Comment:
At the moment, the only value that can change after the initialization is `value` (this is the controlled behaviour). Changing others will be supported in 3.0, so I suggest we don't update `this.suggestionsFn` on the fly. Would you be happy to remove this for now? Also, could you please squash your commits into a single commit?